benchmarks/benchmarks/bench_nexus.py

```diff
@@ -879,45 +899,89 @@ def start(self) -> None:
-                    accessor.bench_result_file(accessor.get_id(schema))
-                    for schema in accessor.schemas
-                ]
-                for env_description, accessor in env_with_accessor
+            env_to_accessor = {
+                env_description: accessor for env_description, accessor in env_with_accessor
             }
+
             with ZipFile(
                 file=RELEASE_DATA / f"{hub_description.key}.zip",
                 mode="w",
                 compression=ZIP_BZIP2,
                 compresslevel=9,
             ) as release_zip:
-                for file_path in chain.from_iterable(env_to_files.values()):
-                    release_zip.write(file_path, arcname=file_path.name)
-
-                release_zip.writestr(
-                    "index.json",
-                    json.dumps(
-                        self._get_index_data(env_to_files),
-                    ),
-                )
+                for env in env_to_accessor:
+                    accessor = env_to_accessor[env]
+                    bench_operator = operator_factory(accessor, sqlite=bool(self.sqlite))
+                    bench_results = bench_operator.get_all_bench_results()
+                    bench_ids = [accessor.get_id(schema) for schema in accessor.schemas]
+                    if isinstance(bench_operator, FileSystemBenchOperator):
+                        self._release_from_files(
+                            bench_operator, release_zip,
+                            bench_ids, bench_results,
+                        )
+                    else:
+                        self._release_from_sqlite(release_zip, bench_ids, bench_results)
 
-    def _get_index_data(self, env_to_files: Mapping[EnvDescription, Iterable[Path]]) -> Dict[str, Any]:
-        return {
-            "env_files": {
-                env_description.key: [
-                    file.name for file in files
-                ]
-                for env_description, files in env_to_files.items()
-            },
-        }
+                if not self.sqlite:
+                    filesys_operator_factory = cast(
+                        Callable[[BenchAccessor, Literal[False]], FileSystemBenchOperator],
+                        operator_factory,
+                    )
+                    release_zip.writestr(
+                        "index.json",
+                        json.dumps(
+                            self._get_index_data(
+                                {
+                                    env_description: [
+                                        filesys_operator_factory(accessor, False).bench_result_file(accessor.get_id(schema))
+                                        for schema in accessor.schemas
+                                    ]
+                                    for env_description, accessor in env_with_accessor
+                                }),
+                        ),
+                    )
```

Comment:
`env_files` written to `index.json` consists of enviroment name and list of output result files. This mapping is actual for any type of working data storage